p3387.tex

```diff
@@ -292,7 +300,8 @@ \subsection{Satisfy the Contracts Prime Directive}
 
 A corollary of this fundamental principle is that adding \tcode{pre} or \tcode{post} to an existing program should not change the compile-time or run-time semantics of that program (see \cite{P2900R8}, Section 3.1, Principles 2 and 3).
 
-Solution 4 violates this principle. While making parameters in \tcode{post} refer to the parameter copy that is part of the coroutine state seems implementable, it will require changes to the semantics of the coroutine, such as extending its lifetime (otherwise, the coroutine state may be destroyed during initial suspend, i.e. before \tcode{post} is checked, and the parameter object along with it). Without making changes to the coroutine semantics, there would be no point at which the parameter copies and the return value necessarily exist at the same time, and therefore no point at which \tcode{post} could be checked.
+???Solution 4 violates this principle. While making parameters in \tcode{post} refer to the parameter copy that is part of the coroutine state seems implementable, it will require changes to the semantics of the coroutine, such as extending its lifetime (otherwise, the coroutine state may be destroyed during initial suspend, i.e. before \tcode{post} is checked, and the parameter object along with it). Without making changes to the coroutine semantics, there would be no point at which the parameter copies and the return value necessarily exist at the same time, and therefore no point at which \tcode{post} could be checked.
+%IDS In my view as an implementer this would be extremely heavy lifting both in the wording and implementation (since it really, really makes a big difference to the nesting of lifetimes).  Of course, it's all "just code" .. and most things could be done if necessary .. but I'd not want to go there if possible.
```

Comment:
I think the fact that it's a semantic change is what makes this a non-starter anyway, so we don't need to dwell on all the reasons it's a bad idea.